extensions/positron-python/src/client/positron/session.ts

```diff
@@ -527,66 +522,62 @@ export class PythonRuntimeSession implements positron.LanguageRuntimeSession, vs
         );
     }
 
+    /**
+     * Start the LSP
+     *
+     * Returns a promise that resolves when the LSP has been activated.
+     */
     async activateLsp(): Promise<void> {
-        // Start LSP for the foreground session only if its been previously stopped
-        if (this._lsp?.state === LspState.stopped) {
-            return this._queue.add(async () => {
-                await this.startLsp();
-            });
-        } else {
-            return undefined;
-        }
-    }
+        return this._queue.add(async () => {
```

Comment:
I moved more of this function and deactivate into the queue because it feels easier to reason about.